### ISSUE TITLE
[feature] sequential (synchronous) action execution

### ISSUE THREAD

**Issue by githorse** [junegunn/fzf#2476]
- [x] I have read through the manual page (`man fzf`)
- [x] I have the latest version of fzf
- [x] I have searched through the existing issues

## Info

When I do something like

``` 
cat my/list | fzf --bind 'ctrl-r:execute(mutate my/list)+reload(cat my/list/)'
```

I have a race condition between the `execute` and the `reload`; if mutating takes too long the reload will not reflect the new state of the list. My kludge right now is to add a sleep to the reload: `reload(sleep 0.5; cat my/list)`. For obvious reasons that's not great. The fix would be to make the actions execute sequentially, one after the other. 

But, of course, 95% of the time I probably *do* want these to execute in parallel, not sequentially. I wonder if there could be an alternate syntax for specifying actions to be run in sequence rather than in parallel? Maybe `>` instead of `+` makes sense? `-` would be another obvious choice.

```
fzf --bind 'ctrl-r:execute(mutate my/list)>reload(cat my/list)'
```

Or maybe I'm missing the best way to do this? I would imagine this is a reasonably common use case. (In my case I want to list open windows, then add a keybinding to close one and update the list.)

**Comment by junegunn** [comment]
No, the actions run sequentially. I think you're mistaken somehow.

```sh
# reload will not start until the previous execute is complete
fzf --bind 'ctrl-r:execute(date > /tmp/date; sleep 2; date >> /tmp/date)+reload(cat /tmp/date)'
```

You can clearly see that fzf doesn't "reload" until the execute action finishes. (And in this case, you probably want to use `execute-silent` instead.)

If you do want to run commands in parallel, you can start one as a background process like so.

```sh
fzf --bind 'ctrl-r:execute-silent((sleep 2; date > /tmp/date) &)+reload(date; echo 'waiting'; sleep 3; cat /tmp/date)'
```

Or you could write a smarter script/program for "reload" that does both in any way you want.

**Comment by junegunn** [comment]
Having said that, you don't even need two actions here.

```sh
fzf --bind 'ctrl-r:reload(mutate my/list; cat my/list)'
```

**Comment by githorse** [comment]
You're right, my bad. I see now that the (real) command I'm using (`wmctrl -c`) returns immediately even if the window hasn't actually closed. 

My problem is that most windows close immediately but some take several seconds. So how do I know how long to wait? I tried something like this:

```
--bind ctrl-k:reload(mutate my/list; sleep 0.1; cat my/list)+reload(sleep 1; cat my/list)+reload(sleep 5; cat my/list)
```

But it seems that only the final reload is executed here, so this won't work. 

I found a much better solution for my particular case, using a `while` loop that keeps running until the window is closed, like this:

```
--bind 'ctrl-k:reload(wmctrl -c {4..}; while wmctrl -l | grep -q {4..}; do sleep 0.1; done; wmctrl -l)' 
```

I guess multiple reload actions might be handy but I'll hold off on proposing that until I have a use case that can't be solved better another way.
